optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/calendar/TimeSlotTableView.java

```diff
@@ -160,11 +163,27 @@ public void addTimeSlot(I shift) {
         T drawable = provider.createDrawable(twoDayViewPresenter, shift, 0);
         uuidToDrawable.put(uuid, drawable);
         drawableToUUID.put(drawable, uuid);
+        shiftToUUID.put(shift, uuid);
         timeSlots.add(drawable);
         updateTimeSlotsFor(shift.getGroupId());
 
     }
 
+    public void updateTimeSlot(I oldShift, I newShift) {
+        if (!oldShift.getGroupId().equals(newShift.getGroupId()) ||
+                !oldShift.getStartTime().equals(newShift.getStartTime()) ||
+                !oldShift.getEndTime().equals(newShift.getEndTime())) {
+            throw new RuntimeException("Old Shift does not exist in the same time slot as New Shift");
```

Comment:
Don't throw RuntimeException, that's a too high level class. Use IllegalStateException, IllegalArgumentException, etc